apps/meteor/app/cloud/server/functions/getWorkspaceLicense.ts

```diff
@@ -51,11 +51,15 @@ const fetchCloudWorkspaceLicensePayload = async ({ token }: { token: string }):
 
 export async function getWorkspaceLicense(): Promise<{ updated: boolean; license: string }> {
 	const currentLicense = await Settings.findOne('Cloud_Workspace_License');
+	// it should never happen, since even if the license is not found, it will return an empty settings
+	if (!currentLicense?._updatedAt) {
+		throw new CloudWorkspaceLicenseError('Failed to retrieve current license');
+	}
 
 	const fromCurrentLicense = async () => {
 		const license = currentLicense?.value as string | undefined;
 		if (license) {
-			callbacks.run('workspaceLicenseChanged', license);
+			await callbacks.run('workspaceLicenseChanged', license);
```

Comment:
Are you sure `await` is necessary, since `workspaceLicenseChanged` is listed under `EventLikeCallbackSignatures`?

Reply:
hm not totally sure, but all callbacks.run returns a promise, if not we should use runAsync